### ISSUE TITLE
Support email-only logins

### ISSUE THREAD

**Issue by amyjko** [wordplaydev/wordplay#628]
### What are you trying to do that you can't?

Managing accounts and passwords is one of the biggest pain points for teachers because students constantly forget their passwords and lose their work. Email-only logins work well for students who have a school email address.

### What is your idea?

This functionality is already implemented, but inactive because of ambiguity about COPPA. The FTC FAQ indicates that not-for-profits are [not required to comply](https://www.ftc.gov/business-guidance/resources/complying-coppa-frequently-asked-questions#A.%20General%20Questions), but schools often do not permit the adoption of tools that don't comply anyway, out of fear. So supporting this is a barrier to adoption.

That said, big players like Scratch use this rationale to gather email addresses for account recovery.

If we did support it, there are a few possible designs:

1. Chooser a username and password OR an email only login
2. Chooser a username, password, AND email

The first option is a bit more complex, since there are multiple ways to log in, but provides a more private option. The second option is simpler, but would require storing the email for existing accounts in some field other than the email field, which is complex internally.

In either case, we'd need to design a way to switch between the two methods as well, update the privacy policy, and communicate throughout exactly what we will use the email address for.

I favor option 1.

### Design

See [approved design](https://github.com/wordplaydev/wordplay/issues/628#issuecomment-2700235279).

**Comment by sdahl4** [comment]
I would like to be assigned to this @amyjko !

**Comment by sdahl4** [comment]
@amyjko 
I unfortunately have to be at work tonight but here are my thoughts on this issue so far:

I also favor option 1. 

It seems like on the login page their is already an option to either use an email only account pass code emailed to the user, what specific design changes are you looking for in regards to switching between the two methods?

I also believe with the ambiguity about COPPA, that we should add a section to the “Rights” section that reads something like this:

For educators regarding regarding student privacy: 
____________________________________________________
If you are an educator that wishes to use Wordplay, that’s great!

We are a non profit organization so according to COPPA , the privacy policy regarding collecting student’s data does not apply to us. 

We still take student’s privacy very seriously, any email we have we only use to help users access their accounts, we don’t use this data for anything else. 

We understand that some educators still may not want to use students data for their accounts, so we have a couple of options in regards to account creation: 

Option 1:
Students can enter a username and password with no email attached, so no worries about using a students email to create an account. 

Option 2: 
Students can create an account with an emailed pass key every time they want to access their account. 

Option 3: 
Educators can also choose to have teacher privileges added to their account, where they can then create and monitor accounts for their students, so no student would have to use their own information to have access to an account.

**Comment by amyjko** [comment]
Thanks for these thoughts! I like the language you've shared here. I'm curious how that compares to the other language you might have found on similar educational platforms?

As for design changes, we currently don't allow the creation of new email-only accounts. So the redesign would have to overhaul the login and account creation pages to make it clear that there are multiple options and what their tradeoffs are.

**Comment by sdahl4** [comment]
@amyjko Yes! I took some inspiration from how scratch emphasized the fact it is a non profit to explain why certain liberties were taken in regards to COPPA. 
I looked around some other no profit coding education websites, mainly ones aimed at younger students, and most either just mentioned that they were a non profit, and some did not mention it at all. 
A thought I have is since most non profit coding education platforms to my knowledge don’t work directly with educators (disregarding scratch), eliminating barriers to use in a classroom setting with clean language around student privacy is most likely not a priority…. so it makes sense that most “big players” are just concerned to make sure their language keeps them legally safe, so I think it makes sense for Wordplays description to go more in depth than other similar EPLS. 
___
We could possibly do a option selection screen for users to select an option before account creation: 

![Image](https://github.com/user-attachments/assets/d267f5e1-90b9-412e-ad4c-3eb074c9ff8b)

Based off their selection the sign up screen would customize their options: 

For an Over Age 13 Programmer selection:
The Screen could show an additional option for an email passkey based account like shown below:

![Image](https://github.com/user-attachments/assets/a1604202-1bfe-4a34-8f48-725ca0b3f800)

For an Under Age 13 Programmer selection:
The screen could continue to just show the username/password option:

![Image](https://github.com/user-attachments/assets/f7bbfb5b-fc00-4b2f-a1d5-8ad26185e614)

For an Educator:
This selection will will have information about the steps to take for creating an Educator account - which has class management privileges? Perhaps a request form if an educator needs to reach out and request that type of account?

**Comment by sdahl4** [comment]
- @amyjko  - Based off of the information from the [GDPR handbook](https://www.gdprhandbook.eu/consent-and-children-data) I found that the EU GDPR has the age cutoff of 16...and children 13-16 can have data collected from them with parental consent. 
- So we could create different versions of the site depending on which country (ie. wordplay.dev/us or wordplay.dev/eu)  if that is even possible...
-  A simpler version would be to modify the age split selection screen to have 
- Over 13 in the US OR over 13 in the EU and Under 13 in the US OR Under 16 in the EU
- What level of verification do you want to engage with? Perhaps a popup with a link to our rules? Here is a popup that models what I was thinking of: 

![Image](https://github.com/user-attachments/assets/d3fb7816-ee99-4119-af11-8af641afe120)
from  [Magento 2 Extensions Documentation](https://docs.mageplaza.com/magento/age-verification/)
I'm guessing we do not want to engage with any actual ID verification. 
- Okay I think we should get rid of the teacher option and instead have a hyperlink saying "Are you an Educator? Click here to learn more" linking to the "teach" page with information about the educator form, on the "over age of 13" selection box. 
- Sure I would love that! My username is sdahl4

**Comment by amyjko** [comment]
I look forward to the mockups!

**Comment by amyjko** [comment]
I also added the teacher privilege to your account, so you can tinker.

**Comment by sdahl4** [comment]
![Image](https://github.com/user-attachments/assets/c1e4dfb2-03df-4b42-90f3-c2acc802a631)

Here are the mockups for the updated account creation screen @amyjko 

We also discussed creating an update to the rights page, which would include:  a statement about the intent of using children's data and a section which distills laws into a brief summary to give insight into why we chose to design our account creation the way we did. 
Please let me know if you know of any other information that would be helpful to add to this!

**Comment by amyjko** [comment]
Wonderful. A few design gaps:

* What does the login page look like with this revised join flow? (Currently, "create an account" is just a link)
* What are the age cutoffs for different countries?
* Should there be a link to the about page to explain why we ask these questions?
* What should the about page say?

**Comment by sdahl4** [comment]
@amyjko The create an account button linking to the flow still works, my only suggestion is to change the appearance of the link to be a larger button, so it is easier for new users to find. The new login page design is attached below. 

The updated flow also has a link to the about day. 

![Image](https://github.com/user-attachments/assets/1b6ed2b0-7f45-40c5-9151-cecb80cf2152)

This flow will act as series of microtranstions, and the orange process bar will progress as the user completes steps. In the second screen (select your country) when the user selects the plus button a drop down menu will expand to display a list of counties the user will select from. 

The orange 'more' in the third screen will link to the rights page. 

The 5 rights foundation, supported by their partner company reset.tech who are supported from two philanthropies – the Sandler Foundation and Luminate (part of The Omidyar Group) created a [comprehensive guide ](https://5rightsfoundation.com/wp-content/uploads/2024/08/Approaches-to-Childrens-Data-Protection-.pdf)to data policies that have specific protections for children's data. 

Based on this data, the age cutoff for creating an email passkey account for all countries should be 13 except for the following countries:

Australia Age cutoff of 18.
Brazil: Age cutoff of 18.
India: Age cutoff of 18.

(quick note that I checked in some new policies for countries in the AU but those policies are mainly focused on storing data that was not conceptually given outside of the countries users live in and don't really apply to this) 

The about page could have the following update, starting with an additional bullet point in the data section with a simpler overview, and below a more detailed explanation aimed at older users. 

- your email: When you create an account using an email passkey, we keep your email so we know there's an account connected to it. This helps us send you the login passkey you need. If you’re wondering why we ask things like what country you live in or how old you are, it’s because different countries have rules about keeping your information safe. These rules help make sure websites like ours collect your information in the best way possible.

Data collection is something we only do when necessary, and we only use it to help users access their account. Many countries emphasize that data collection should be done in the users best interest, and we comply with that request by only using your data to help you access your account. You have complete control over your own data, you can erase your account and your data will automatically be deleted from our database.

**Comment by amyjko** [comment]
Thanks for this iteration @sdahl4, I think this is ready. There are some minor design questions — where does the list of countries come from, when we localize them, do we use English names or localized names — but I think I'm ready to approve this, and save some of those smaller details for the implementation phase. Congratulations, and nice work on this!

**Comment by sdahl4** [comment]
@amyjko I was looking at [this](https://www.worldatlas.com/geography/names-of-countries-in-their-own-languages.html) list of countries.  I suggest we should have them in the format: Localized name (if applicable)/English name.

**Comment by amyjko** [comment]
That seems like a reasonable list! We could integrate it in with the region information we already have for language tags.

**Comment by shreyam47** [comment]
@amyjko cleanup - inactivity
